generator/search_index_builder.cpp

```diff
@@ -443,23 +442,41 @@ void BuildAddressTable(FilesContainerR & container, Writer & writer, uint32_t th
 
   // Flush results to disk.
   {
-    FixedBitsDDVector<3, FileReader>::Builder<Writer> building2Street(writer);
-    for (auto i : results)
+    // Code corresponds HouseToStreetTable decoding.
+    MapUint32ToValueBuilder<uint32_t> builder;
+    uint32_t houseToStreetCount = 0;
+    for (size_t i = 0; i < results.size(); ++i)
     {
-      if (i == kEmptyResult)
-        building2Street.PushBackUndefined();
-      else
-        building2Street.PushBack(i);
+      if (results[i] != kEmptyResult)
+      {
+        builder.Put(base::asserted_cast<uint32_t>(i), results[i]);
+        ++houseToStreetCount;
+      }
     }
 
-    LOG(LINFO, ("Address: Building -> Street (opt, all)", building2Street.GetCount()));
+    // Each street id is encoded as delta from some prediction.
+    // First street id in the block encoded as VarUint, all other street ids in the block
+    // encoded as VarInt delta from previous id.
+    auto const writeBlockCallback = [&](Writer & w, vector<uint32_t>::const_iterator begin,
+                                        vector<uint32_t>::const_iterator end) {
+      WriteVarUint(w, *begin);
```

Comment:
What if begin == end ?

Reply:
check added here and inside MapUint32ToValue